pkg/api/api.go

```diff
@@ -237,47 +253,66 @@ func Series(ctx context.Context, client promapi.Client, matches []string, startT
 	q.Set("start", formatTime(startTime))
 	q.Set("end", formatTime(endTime))
 
-	_, body, warnings, err := doGetFallback(ctx, client, u, q, cache) //nolint:bodyclose
+	resp, body, warnings, err := doGetFallback(ctx, client, u, q, cache) //nolint:bodyclose
 	if err != nil {
-		return nil, warnings, err
+		return nil, resp.StatusCode, warnings, err
 	}
 
 	var mset []model.LabelSet
-	return mset, warnings, json.Unmarshal(body, &mset)
+
+	return mset, resp.StatusCode, warnings, json.Unmarshal(body, &mset)
 }
 
 func LabelNames(ctx context.Context, client promapi.Client, startTime time.Time, endTime time.Time,
-	cache bool) ([]string, promapiv1.Warnings, error) {
+	cache bool) ([]string, int, promapiv1.Warnings, error) {
 	u := client.URL(epLabels, nil)
 	q := u.Query()
 	q.Set("start", formatTime(startTime))
 	q.Set("end", formatTime(endTime))
 
 	u.RawQuery = q.Encode()
 
-	_, body, warnings, err := doGetFallback(ctx, client, u, q, cache) //nolint:bodyclose
+	resp, body, warnings, err := doGetFallback(ctx, client, u, q, cache) //nolint:bodyclose
 	if err != nil {
-		return nil, warnings, err
+		return nil, resp.StatusCode, warnings, err
 	}
+
 	var labelNames []string
-	return labelNames, warnings, json.Unmarshal(body, &labelNames)
+
+	if resp == nil {
+		// Unknown error.
+		return labelNames, 520, warnings, json.Unmarshal(body, &labelNames)
```

Comment:
```suggestion
		return labelNames, 0, warnings, json.Unmarshal(body, &labelNames)
```

Same suggestion as above.

Reply:
we can have 0 as the code for any errors prior to making HTTP call. Here, we would have made the HTTP call, could not parse response and can assume that server has returned an unknown error (https://en.wikipedia.org/wiki/List_of_HTTP_status_codes)